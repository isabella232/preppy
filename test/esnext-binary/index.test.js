/* global __dirname */
import { readFile } from "fs"
import { resolve } from "path"
import pify from "pify"
import rimraf from "rimraf"

import preppy from "../../src/index"

const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)

jest.setTimeout(20000)

test("Binary from ESNext", async () => {
  await lazyDelete(resolve(__dirname, "./bin"))

  await preppy({
    root: __dirname
  })

  expect(
    await lazyRead(resolve(__dirname, "bin/mycli.js"), "utf8")
  ).toMatchSnapshot("cli")
})
