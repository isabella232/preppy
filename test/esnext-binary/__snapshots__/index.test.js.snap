// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binary from ESNext: cli 1`] = `
"#!/usr/bin/env node

/*! esnext-binary v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

var lodash = require('lodash');

var realWords = \\"Eine dicke Kuh lacht\\";

console.log(\\"Magic:\\", lodash.snakeCase(realWords));
//# sourceMappingURL=mycli.js.map
"
`;
